### PR TITLE
EC34 replaced by EC35 : PHP implementation 

### PR DESCRIPTION
**Pull request by dedece35**
3nd PR for TODO tasks mentionned in PR https://github.com/green-code-initiative/ecoCode/pull/128

waiting for new release of rules-specifications before merging this PR ... see PR https://github.com/green-code-initiative/ecoCode/pull/236 : once merged and a new manual release fo ecocode-rules-specifications done, use it on pom.xml

### CODE REVIEW

File: src/main/java/fr/greencodeinitiative/php/checks/AvoidTryCatchWithFileOpenedCheck.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            AssignmentExpressionTree assignTree = (AssignmentExpressionTree) exprTree;
+            if (assignTree.value().is(Tree.Kind.FUNCTION_CALL)) { // et si "function call"
+                visitCallExpression((FunctionCallTree) assignTree.value());
+            }
+        }
+    }
+
+    private void visitCallExpression(FunctionCallTree functionCall){
+        String funcName = getFunctionNameFromCallExpression(functionCall);
+        if (funcName.startsWith("PDF_open")
```
Comment (utarwyn) [inline src/main/java/fr/greencodeinitiative/php/checks/AvoidTryCatchWithFileOpenedCheck.java:146]: This method seems to be very old in PHP (I only see it in PHP 4 which is outdated). Do we need to handle it in this rule?

File: src/main/java/fr/greencodeinitiative/php/checks/AvoidTryCatchWithFileOpenedCheck.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                || "fopen".equals(funcName)
+                || "readfile".equals(funcName)) {
```
Comment (utarwyn) [inline src/main/java/fr/greencodeinitiative/php/checks/AvoidTryCatchWithFileOpenedCheck.java:148]: Methods are not case-sensitive in PHP, so maybe we could use `equalsIgnoreCase` instead?
The idea is to handle more cases if some developers use uppercase

**Review by utarwyn — CHANGES_REQUESTED** [review]
There seems to be a problem with the coverage... can you have a look?

File: src/main/java/fr/greencodeinitiative/php/checks/AvoidTryCatchWithFileOpenedCheck.java (hunk 1 above)
Comment (dedece35) [reply src/main/java/fr/greencodeinitiative/php/checks/AvoidTryCatchWithFileOpenedCheck.java:146]: indeed, I llooked for all PHP methods to open file ... thus I added it. But I can delete it if you prefer.

File: src/main/java/fr/greencodeinitiative/php/checks/AvoidTryCatchWithFileOpenedCheck.java (hunk 2 above)
Comment (dedece35) [reply src/main/java/fr/greencodeinitiative/php/checks/AvoidTryCatchWithFileOpenedCheck.java]: please check https://github.com/green-code-initiative/ecoCode-php/pull/10/commits/ed98eb7a9d661105a4301c09ebff951b390af073

**Comment by dedece35** [comment]
> There seems to be a problem with the coverage... can you have a look?

After analysis, I realize that the coverage is 0% also on main branch on all other code.
I think there is a problem with coverage on PHP repository.
I continue investigating.

**Pull request merged**